# Exercise on practicing with removing outliers using percentile based on price per night for a given apartment/home

In [1]:
# import libraries

import numpy as np
import pandas as pd

In [2]:
df =  pd.read_csv('/kaggle/input/new-york-city-airbnb-open-data/AB_NYC_2019.csv')

In [3]:
df.shape

(48895, 16)

In [4]:
df['price'].describe()

count    48895.000000
mean       152.720687
std        240.154170
min          0.000000
25%         69.000000
50%        106.000000
75%        175.000000
max      10000.000000
Name: price, dtype: float64

In [5]:
df['price'].quantile(0.0001)

0.0

In [6]:
df['price'].quantile(0.99)

799.0

In [7]:
# We need to remove zero from prices and the high prices from the dataset

min_percentile = df['price'].quantile(0.0001)
max_percentile = df['price'].quantile(0.99)

In [8]:
new_df = df[df['price'] > min_percentile]
new_df = df[df['price'] < max_percentile]

In [9]:
new_df.shape

(48403, 16)

In [10]:
new_df['price'].describe()

count    48403.000000
mean       137.297936
std        103.021327
min          0.000000
25%         69.000000
50%        105.000000
75%        175.000000
max        795.000000
Name: price, dtype: float64

**So, by comparing the shapes of old and new dataframe we found ->  48895 - 48403 = 492 outliers !**